packages/core/parcel-bundler/src/assets/TypeScriptAsset.js

```diff
@@ -1,12 +1,53 @@
+const path = require('path');
 const Asset = require('../Asset');
 const localRequire = require('../utils/localRequire');
+const config = require('../utils/config');
+const TSCONFIG_FILENAMES = ['tsconfig.json'];
 
 class TypeScriptAsset extends Asset {
   constructor(name, options) {
     super(name, options);
     this.type = 'js';
   }
 
+  async readConfigFile(filepath = this.name, seenConfigPath = new Set()) {
+    const tsconfigPath = await config.resolve(filepath, TSCONFIG_FILENAMES);
+    if (!tsconfigPath) return;
+    if (seenConfigPath.has(tsconfigPath)) {
+      // Loop detected
+      // This shouldn't happen, but let's silently ignore it?
+      return;
+    }
+    seenConfigPath.add(tsconfigPath);
+
+    // Add as a dependency so it is added to the watcher and invalidates
+    // this asset when the config changes.
+    this.addDependency(tsconfigPath, {includedInParent: true});
+
+    const tsconfig = await config.load(filepath, TSCONFIG_FILENAMES);
+    if (tsconfig.extends && typeof tsconfig.extends === 'string') {
+      const parentConfigPath = path.join(
+        path.dirname(tsconfigPath),
+        tsconfig.extends
+      );
+      const parentConfig = await this.readConfigFile(
+        parentConfigPath,
+        seenConfigPath
+      );
+      const compilerOptions = Object.assign(
+        {},
+        parentConfig && parentConfig.compilerOptions,
+        tsconfig.compilerOptions
+      );
+      const mergedConfig = Object.assign({}, parentConfig, tsconfig, {
+        compilerOptions
+      });
+      delete mergedConfig.extends;
+      return mergedConfig;
+    }
+    return tsconfig;
+  }
+
```

Comment:
What do you think of the way #2150 approached this? i.e. using the typescript compiler infrastructure to do the config loading

Reply:
Yeah but also the config format could also change... I'm not sure we should be duplicating the extension mechanism for that reason.

Reply:
Sure thing. I can revert back to what I had with Typescript implementation in this commit (https://github.com/parcel-bundler/parcel/pull/2269/commits/dcc8ec005f2f6545844ba4aa40c3e752ca1450d2). I will fix it tomorrow :) 

But, I have no clue why this PR is failing tests..